PrebidMobile/PrebidMobile-core/src/main/java/org/prebid/mobile/AdUnit.java

```diff
@@ -200,19 +200,10 @@ public void fetchDemand(OnFetchDemandResult listener) {
         allowNullableAdObject = true;
 
         fetchDemand(null, resultCode -> {
-            BidInfo bidInfo;
-            if (bidResponse != null) {
-                bidInfo = new BidInfo(resultCode, bidResponse.getTargeting());
-                if (resultCode == ResultCode.SUCCESS) {
-                    boolean isNative = configuration.getNativeConfiguration() != null;
-                    if (isNative) {
-                        String cacheId = CacheManager.save(bidResponse.getWinningBidJson());
-                        Util.saveCacheId(cacheId, adObject);
-                        bidInfo.setNativeResult(cacheId, bidResponse.getExpirationTimeSeconds());
-                    }
-                }
-            } else {
-                bidInfo = new BidInfo(resultCode, null);
+            BidInfo bidInfo = BidInfo.create(resultCode, bidResponse);
+            boolean isNative = configuration.getNativeConfiguration() != null;
```

Comment:
Can we move the addition of native information into the `BidInfo.create` method? It looks essential for the constructor of the BidInfo. Just pass the `configuration` into BidInfo.create.